packages/features/bookings/lib/BookingEmailSmsHandler.ts

```diff
@@ -0,0 +1,304 @@
+          timeFormat: getTimeFormatStringFromUserTimeFormat(originalRescheduledBooking.user.timeFormat),
+          name: originalRescheduledBooking.user.name || "",
+          language: { translate, locale: originalRescheduledBooking.user.locale ?? "en" },
+        });
+
+        if (changedOrganizer) {
+          cancelledRRHostEvt.title = originalRescheduledBooking.title;
+          cancelledRRHostEvt.startTime =
+            dayjs(originalRescheduledBooking?.startTime).utc().format() || copyEventAdditionalInfo.startTime;
+          cancelledRRHostEvt.endTime =
+            dayjs(originalRescheduledBooking?.endTime).utc().format() || copyEventAdditionalInfo.endTime;
+          cancelledRRHostEvt.organizer = {
+            email: originalOrganizer.email,
+            name: originalOrganizer.name || "",
+            timeZone: originalOrganizer.timeZone,
+            language: { translate, locale: originalOrganizer.locale || "en" },
+          };
+        }
+      }
+
+      const newBookingMemberEmails: Person[] = [
+        ...(copyEvent.team?.members || []),
+        copyEvent.organizer,
+        ...copyEvent.attendees,
+      ];
+
+      const matchOriginalMemberWithNewMember = (originalMember: Person, newMember: Person) =>
+        originalMember.email === newMember.email;
+
+      const newBookedMembers = newBookingMemberEmails.filter(
+        (member) => !originalBookingMemberEmails.some((om) => matchOriginalMemberWithNewMember(om, member))
+      );
+      const cancelledMembers = originalBookingMemberEmails.filter(
+        (member) => !newBookingMemberEmails.some((nm) => matchOriginalMemberWithNewMember(member, nm))
+      );
+      const rescheduledMembers = newBookingMemberEmails.filter((member) =>
+        originalBookingMemberEmails.some((om) => matchOriginalMemberWithNewMember(om, member))
+      );
+
+      try {
+        await sendRoundRobinRescheduledEmailsAndSMS(
+          { ...copyEventAdditionalInfo, iCalUID: evt.iCalUID },
+          rescheduledMembers,
+          metadata
+        );
+        await sendRoundRobinScheduledEmailsAndSMS({
+          calEvent: copyEventAdditionalInfo,
+          members: newBookedMembers,
+          eventTypeMetadata: metadata,
+        });
+        const reassignedTo = users.find(
+          (user) => !user.isFixed && newBookedMembers.some((member) => member.email === user.email)
+        );
+        await sendRoundRobinCancelledEmailsAndSMS(
+          cancelledRRHostEvt,
+          cancelledMembers,
+          metadata,
+          reassignedTo
+            ? {
+                name: reassignedTo.name,
+                email: reassignedTo.email,
+                ...(isRescheduledByBooker && { reason: "Booker Rescheduled" }),
+              }
+            : undefined
+        );
+      } catch (err) {
+        this.log.error("Failed to send rescheduled round robin event related emails", err);
+      }
+    } else {
+      await sendRescheduledEmailsAndSMS(
```

Comment:
Should we wrap the imported sending functions e.g. `sendRescheduledEmailsAndSMS` in class functions e.g. `BookingEmailSmsHandler.sendRescheduledEmailsAndSMS` that then calls `sendRescheduledEmailsAndSMS` ? That would allow us in tests to mock `BookingEmailSmsHandler.sendRescheduledEmailsAndSMS` if we don't want that the email is actually sent during testing or we could expect that it was called.

We can implement mocks on the individual email level like we do in e2e tests in v2
```
jest
  .spyOn(AttendeeRescheduledEmail.prototype, "getHtml")
  .mockImplementation(() => Promise.resolve("<p>email</p>"));
```

but having class functions that we can mock / track could be useful for integration tests in the future as they abstract the actual call to 3rd party.